src/main/java/com/edmunds/tools/databricks/maven/util/SettingsInitializer.java

```diff
@@ -0,0 +1,13 @@
+package com.edmunds.tools.databricks.maven.util;
+
+import com.edmunds.tools.databricks.maven.model.BaseModel;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import org.apache.maven.plugin.MojoExecutionException;
+
+public interface SettingsInitializer<T extends BaseModel, D> {
```

Comment:
can you add a javadoc that explains the intent of this class

Reply:
Done